src/visualization/plot_types.jl

```diff
@@ -0,0 +1,176 @@
+"""
+struct PlotData2D{Coordinates, Data, VariableNames, Vertices} <: AbstractPlotData{2}
+  x::Coordinates
+  y::Coordinates
+  data::Data
+  variable_names::VariableNames
+  mesh_vertices_x::Vertices
+  mesh_vertices_y::Vertices
+  orientation_x::Int
+  orientation_y::Int
+end
+
+# holds plotting information for UnstructuredMesh2D and DGMulti-compatible meshes
+struct UnstructuredPlotData2D{DataType, FaceDataType, VariableNames, PlottingTriangulation, RealT} <: AbstractPlotData{2}
+  x::Array{RealT, 2} # physical nodal coordinates, size (num_plotting_nodes x num_elements)
+  y::Array{RealT, 2}
+  data::DataType
+  t::PlottingTriangulation
+  x_face::Array{RealT, 2}
+  y_face::Array{RealT, 2}
+  face_data::FaceDataType
+  variable_names::VariableNames
+end
+
+# Show only a truncated output for convenience (the full data does not make sense)
+function Base.show(io::IO, pd::PlotData2D)
+  @nospecialize pd # reduce precompilation time
+
+  print(io, "PlotData2D{",
+            typeof(pd.x), ",",
+            typeof(pd.data), ",",
+            typeof(pd.variable_names), ",",
+            typeof(pd.mesh_vertices_x),
+            "}(<x>, <y>, <data>, <variable_names>, <mesh_vertices_x>, <mesh_vertices_y>)")
+end
+
+
+# Show only a truncated output for convenience (the full data does not make sense)
+function Base.show(io::IO, pd::UnstructuredPlotData2D)
+  @nospecialize pd # reduce precompilation time
+
+  print(io, "UnstructuredPlotData2D{",
+            typeof(pd.x), ", ",
+            typeof(pd.u), ", ",
+            typeof(pd.xf), ", ",
+            typeof(pd.uf), ", ",
+            typeof(pd.variable_names),
+            "}(<x>, <y>, <data>, <plot_triangulation>, <x_face>, <y_face>, <face_data>, <variable_names>)")
+end
+
+"""
+    PlotData1D
+
+Holds all relevant data for creating 1D plots of multiple solution variables and to visualize the
+mesh.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct PlotData1D{Coordinates, Data, VariableNames, Vertices} <: AbstractPlotData{1}
+  x::Coordinates
+  data::Data
+  variable_names::VariableNames
+  mesh_vertices_x::Vertices
+  orientation_x::Integer
+end
+
+# Show only a truncated output for convenience (the full data does not make sense)
+function Base.show(io::IO, pd::PlotData1D)
+  print(io, "PlotData1D{",
+            typeof(pd.x), ",",
+            typeof(pd.data), ",",
+            typeof(pd.variable_names), ",",
+            typeof(pd.mesh_vertices_x),
+            "}(<x>, <data>, <variable_names>, <mesh_vertices_x>)")
+end
+
+# Auxiliary data structure for visualizing a single variable
+#
+# Note: This is an experimental feature and may be changed in future releases without notice.
+struct PlotDataSeries{PD<:AbstractPlotData}
+  plot_data::PD
+  variable_id::Int
+end
+
+const PlotDataSeries1D{PD} = PlotDataSeries{PD} where {PD <: AbstractPlotData{1}}
```

Comment:
Is this definition (and the one below) even used anymore?

Reply:
There were a few places in the tests and recipes, but I went ahead and removed those. 

Fewer type aliases would be clearer IMO.